### DISCUSSION TITLE
How to Clean the Removal and Installation of nerdctl？

### DISCUSSION THREAD

**Discussion by liangyoufeng** [containerd/nerdctl#3030] [Q&A] [answered]
Hello, may I ask how to uninstall it? I have fully installed it. How do I want to uninstall it?

Full installation of nerdctl command：
Extract the archive to a path like /usr/local or ~/.local .
tar Cxzvvf /usr/local nerdctl-full-1.7.6-linux-amd64.tar.gz

Custom installation nerdctl command：
mkdir -p ~/.local
tar Cxzvvf ~/.local nerdctl-full-1.7.6-linux-amd64.tar.gz
tar (child): nerdctl-full-1.7.6-linux-amd64.tar.gz: Cannot open: No such file or directory
tar (child): Error is not recoverable: exiting now
tar: Child returned status 2
tar: Error is not recoverable: exiting now

Returning to the error message above, how can I cleanly delete nerdctl?

**Comment by AkihiroSuda** [comment, marked as answer]
> tar (child): nerdctl-full-1.7.6-linux-amd64.tar.gz: Cannot open: No such file or directory

In this case nothing was installed at all

**Reply by liangyoufeng** [reply]
Sorry, it's possible that I didn't describe it properly. The above is the installation method. The first one is that it can be installed and used according to the tutorial. The second one is that there is an error message that cannot be installed by decompressing. I would like to know how to clean up nerdctl

**Reply by AkihiroSuda** [reply]
You can run `tar tzvf` to check the installed files and remove them

**Reply by AkihiroSuda** [reply]
You'll also want to remove
- `~/.local/share/{containerd, nerdctl, cni, buildkit}`
- `/var/lib/{containerd, nerdctl, cni, buildkit}`

**Reply by liangyoufeng** [reply]
According to the command tar tzvf nerdctl full 1.7.6 Linux amd64. tar. gz you provided to detect the corresponding file, it can be deleted. Would there be an overwrite deletion if there are other services referencing it?
For example, if I first install nerdctl full, then install k3s and perform a cleanup, it seems to affect it.
Unfortunately, I am unable to execute the previous step
Custom installation nerdctl command：
mkdir -p ~/.local
tar Cxzvvf ~/.local nerdctl-full-1.7.6-linux-amd64.tar.gz
There is no such step
~/. local/share/{containerd, nerdctl, cni, buildkit}
Additionally, this directory only has the containerd dire

**Comment by raylu** [comment]
on my debian system, I also had to remove the nerdctl0 interface
```
sudo apt purge containerd
rm -r ~/.local/share/nerdctl/
sudo rm -r /etc/nerdctl
sudo /var/lib/{containerd,nerdctl,cni,buildkit}
sudo rm /usr/local/buildkit* /usr/local/nerdctl
sudo ip link delete nerdctl
```
also don't forget to remove your CNI plugins
